test/functional/test_runner.py

```diff
@@ -173,7 +173,7 @@ def main():
     args, unknown_args = parser.parse_known_args()
 
     # Create a set to store arguments and create the passon string
-    tests = set(arg for arg in unknown_args if arg[:2] != "--")
+    tests = [arg for arg in unknown_args if arg[:2] != "--"]
```

Comment:
This is great functionality to allow tests to be run multiple times. It might be useful to edit https://github.com/MarcoFalke/bitcoin/blob/Mf1705-qaWarnTestNotFound/test/README.md#functional-tests  to specify that you can, in fact, run the same test a few times to catch intermittent errors.